dartagnan/src/main/java/com/dat3m/dartagnan/program/event/core/MemEvent.java

```diff
@@ -68,6 +68,11 @@ public String getMo(){
         return mo;
     }
 
+    public void setMo(String mo){
+        this.mo = mo;
+        addFilters(mo);
+    }
```

Comment:
I think it is safer to remove the old `mo` from the `filters` before setting the new `mo`.
This way, calling code doesn't have to remove any filters manually (e.g. `AtomicityPropagation`).
Also, since we allow `NULL` values, you should check that the new value is not null before adding it to the filters.

Reply:
The question is: do we want to allow calling `e.setMo(null)` (meaning we would guard `addFilters` by the `!= null` check) or not (and thus use `Preconditions.checkNotNull`)?

Reply:
The problem that I see is that the default value of `mo` is `null` which seems to be correctly treated in our code.
So it suggests that `null` is valid. In principle, I'm all for removing `null`, but would you then put e.g. `NONATOMIC` as the default value? Or would you keep the default value `null`, but won't allow the value to be reset to `null` once it was changed?

Reply:
I'm fine with that

Reply:
You might still want to do an `null`-check on the old mo-value before removing it from the filters.
I think right now it may not be a problem, but some set implementations may throw exceptions when trying to remove `null`.